### PR TITLE
Merge SOC-Process-Framework Content to Master

### PR DESCRIPTION
**Pull request by rinure-msft**
Required items, please complete
   
   Change(s):
   - Adding the rest of the SOC-Process-Framework Content to the Solution Folder.

   Reason for Change(s):
   - Only one Watchlist was added last PR. This is to complete the Content Merge.

   Version Updated:
   - Version 2.0 of SOC Process Framework

   Testing Completed:
   - Yes - Testing has been completed within the GBB Lab.

   Checked that the validations are passing and have addressed any issues that are present:
   - Not Yet - (I will)

### CODE REVIEW

**Comment by vmanojreddy** [comment]
@anki-narravula please review Playbooks and Watchlists, thanks.

**Comment by elforb** [comment]
Watchlists look fine here.

**Review by v-webr — COMMENTED** [review]
Please review https://github.com/Azure/Azure-Sentinel/tree/master/Workbooks on the use of workbooks and their intended purpose.

The workbooks included in this PR do not pass a series of checklist items that are typically required of a workbook, such as:

- Image previews for various accessibility settings defined under the "previewImagesFileNames" object parameter. These images are to show that you're workbook is accessible to people with various visual impairments.
- A workbook should include at least 4 charts or graphs. I did not find any charts or graphs within these workbooks, only images.
- The 'provider' object parameter should be defined; in this case, 'Microsoft'

There are more items that must meet the workbook criteria. My main question is if this is actually intended as a workbook, as it seems more informative rather than to display aggregated log data.

**Review by v-webr — CHANGES_REQUESTED** [review]
Please see above comment. I'm going to block this for now.

**Review by v-webr — APPROVED** [review]

**Review by anki-narravula — COMMENTED** [review]
Hi @rinure-msft - Thanks for your contribution to Sentinel community. I have reviewed the [playbook](Solutions/SOC-Process-Framework/Playbooks/Get-SOCActions/azuredeploy.json)and I have couple of suggestions

1. We advice to use Incident trigger directly (it is available), rather than reading it from the alert.
2. We can use managed identity for sentinel connections so that there will be seamless experience for users post development.  
3. You can refer template for using managed identity and also for incident trigger  - [https://github.com/Azure/Azure-Sentinel/blob/master/Playbooks/.template/incident-trigger/azuredeploy.json](https://github.com/Azure/Azure-Sentinel/blob/master/Playbooks/.template/incident-trigger/azuredeploy.json)

**Comment by rinure-msft** [comment]
> Hi @rinure-msft - Thanks for your contribution to Sentinel community. I have reviewed the [playbook](Solutions/SOC-Process-Framework/Playbooks/Get-SOCActions/azuredeploy.json)and I have couple of suggestions
> 
> 1. We advice to use Incident trigger directly (it is available), rather than reading it from the alert.
> 2. We can use managed identity for sentinel connections so that there will be seamless experience for users post development.
> 3. You can refer template for using managed identity and also for incident trigger  - https://github.com/Azure/Azure-Sentinel/blob/master/Playbooks/.template/incident-trigger/azuredeploy.json

Hi @anki-narravula, thank you for your suggestions.

This playbook was asked for by customers to be triggered from alerts so they can decide when to add this playbook to Alerts during the Alert design process of the framework. This way Analysts know what steps to take when an Alert has been onboarded rather than by an Incident. To that point, this playbook uses a Watchlist called "SOCRA" that contains the Alert name mapping the Playbook uses as a lookup to know what prescripted steps to apply so the Analyst knows how to proceed when they start to Triage and Investigate. This is by design and gives SOCs the ability to onboard new Alerts with custom steps prescribed in the Watchlist. I hope this helps you understand why this playbook was designed using an Alert Trigger. This playbook and its associated watchlist has already been published to GitHub as stand-alone content and many customers are already using it. By adding this Playbook to this Solution, we are giving customers the ability to install this along with its associated Watchlists and Workbooks as a single installation rather than having to install each of these pieces individually. Thank you for helping us achieve this.

**Comment by vmanojreddy** [comment]
@anki-narravula please review  Workflows. thanks!!!

**Comment by vmanojreddy** [comment]
@anki-narravula please review Workflows. thanks!!!

**Review by anki-narravula — DISMISSED** [review]
Playbooks look good. Approved.

**Comment by anki-narravula** [comment]
> > Hi @rinure-msft - Thanks for your contribution to Sentinel community. I have reviewed the [playbook](Solutions/SOC-Process-Framework/Playbooks/Get-SOCActions/azuredeploy.json)and I have couple of suggestions
> > 
> > 1. We advice to use Incident trigger directly (it is available), rather than reading it from the alert.
> > 2. We can use managed identity for sentinel connections so that there will be seamless experience for users post development.
> > 3. You can refer template for using managed identity and also for incident trigger  - https://github.com/Azure/Azure-Sentinel/blob/master/Playbooks/.template/incident-trigger/azuredeploy.json
> 
> Hi @anki-narravula, thank you for your suggestions.
> 
> This playbook was asked for by customers to be triggered from alerts so they can decide when to add this playbook to Alerts during the Alert design process of the framework. This way Analysts know what steps to take when an Alert has been onboarded rather than by an Incident. To that point, this playbook uses a Watchlist called "SOCRA" that contains the Alert name mapping the Playbook uses as a lookup to know what prescripted steps to apply so the Analyst knows how to proceed when they start to Triage and Investigate. This is by design and gives SOCs the ability to onboard new Alerts with custom steps prescribed in the Watchlist. I hope this helps you understand why this playbook was designed using an Alert Trigger. This playbook and its associated watchlist has already been published to GitHub as stand-alone content and many customers are already using it. By adding this Playbook to this Solution, we are giving customers the ability to install this along with its associated Watchlists and Workbooks as a single installation rather than having to install each of these pieces individually. Thank you for helping us achieve this.

Hi @rinure-msft  thanks for the details. In this case we can go ahead with the existing version then

File: Solutions/SOC-Process-Framework/Watchlists/SOC-Recommended-Actions/SocRA.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
...itigate the breach. Stop suspicious processes, isolate affected machines, decommission compromised accounts or reset passwords, block IP addresses and URLs, and install security updates.,9. Escalate - Contact your incident response team, or contact Microsoft support for investigation and remediation services.,3/18/2021Suspicious Attachment Opened,1. Validate the alert.,2. Inspect the attachment. Review the process that opened it and its behaviors.,3. Check for other suspicious activities in the machine timeline.,4. Locate unfamiliar processes in the process tree. Check files for prevalence, their locations, and digital signatures.,5. Submit relevant files for deep analysis and review file behaviors.,6. Identify unusual system activity with system owners.,7. Scope the incident. Find related machines, network addresses, and files in the incident graph.,8. Contain and mitigate the breach. Stop suspicious processes, isolate affected machines, decommission compromised accounts or reset passwords, block IP addresses and URLs, and install security updates.,9. Escalate - Contact your incident response team, or contact Microsoft support for investigation and remediation services.,2/24/2021"
```
Comment (vmanojreddy) [inline Solutions/SOC-Process-Framework/Watchlists/SOC-Recommended-Actions/SocRA.json:29]: @rinure-msft  rawContent not be in correct format at time deploying we are facing issue. Can you please check and update it. For your reference I'm attaching screenshot. thanks!!!

image
image

**Review by vmanojreddy — CHANGES_REQUESTED** [review]

**Review by vmanojreddy — CHANGES_REQUESTED** [review]

**Comment by vmanojreddy** [comment]
@rinure-msft can you please update the request changes. thanks!!!

File: Solutions/SOC-Process-Framework/Watchlists/SOC-Recommended-Actions/SocRA.json (hunk 1 above)
Comment (rinure-msft) [reply Solutions/SOC-Process-Framework/Watchlists/SOC-Recommended-Actions/SocRA.json:29]: I am having trouble validating the format. I will resubmit today so you can test.

**Comment by vmanojreddy** [comment]
@rinure-msft can you please update the request changes. thanks!!!

**Review by vmanojreddy — APPROVED** [review]

**Pull request merged**